lib/deltachat.ts

```diff
@@ -574,9 +574,12 @@ export class DeltaChat extends EventEmitter {
 
   // TODO this should most likely be async, see
   // https://c.delta.chat/classdc__context__t.html#ae49176cbc26d4d40d52de4f5301d1fa7
-  joinSecurejoin (qrCode:string) {
+  joinSecurejoin (qrCode:string, callback: (result: number) => void) {
     debug(`joinSecurejoin ${qrCode}`)
-    return binding.dcn_join_secure(this.dcn_context, qrCode)
+    if (!callback || typeof callback !== 'function') {
```

Comment:
This check is not really needed if we use typescript, but not sure if we should keep vanilla js compatibility/conveniance. Not having such checks/guards for sure makes using deltachat-node without typescript a bit more a pain in the ass.